ankimorphs/filters/am_highlight_morphs.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+
+from anki.template import TemplateRenderContext
+
+from ..ankimorphs_config import (
+    AnkiMorphsConfig,
+    AnkiMorphsConfigFilter,
+    get_matching_read_filter,
+)
+from ..ankimorphs_db import AnkiMorphsDB
+from ..ankimorphs_globals import EXTRA_FIELD_HIGHLIGHTED
+from ..morpheme import Morpheme
+from ..morphemizers import spacy_wrapper
+from ..morphemizers.morphemizer import (
+    Morphemizer,
+    SpacyMorphemizer,
+    get_all_morphemizers,
+    get_morphemizer_by_description,
+)
+from ..text_highlighting import get_highlighted_text
+from ..text_preprocessing import (
+    get_processed_morphemizer_morphs,
+    get_processed_spacy_morphs,
+)
+
+
+def am_highlight_morphs(
+    field_text: str,
+    field_name: str,
+    filter_name: str,
+    context: TemplateRenderContext,
+) -> str:
+    """Use morph learning progress to decorate the morphemes in the supplied text.
+    Adds css classes to the output that can be styled in the card."""
+
+    if filter_name != "am-highlight-morphs":
+        return field_text
+
+    morphemizers: list[Morphemizer] = get_all_morphemizers()
+
+    if morphemizers is None:
+        return field_text
+
+    note = context.note()
+
+    am_config_filter: AnkiMorphsConfigFilter | None = get_matching_read_filter(note)
+
+    if (
+        am_config_filter
+        and am_config_filter.field == field_name
+        and EXTRA_FIELD_HIGHLIGHTED in " ".join(note.keys())
+    ):
+        return note[EXTRA_FIELD_HIGHLIGHTED]
+
+    morphemizer: Morphemizer | None = (
+        get_morphemizer_by_description(am_config_filter.morphemizer_description)
+        if am_config_filter is not None
+        else morphemizers[-1]
+    )
```

Comment:
If there is no config for this note type, use the last added morpemizer. (they are added in increasing fidelity)

Reply:
Removed this now it's required to create a "read" note config with a morpemizer for each note type that will use this filter.